mne/connectivity/cfc.py

```diff
@@ -0,0 +1,497 @@
+# Authors: Praveen Sripad <praveen.sripad@rwth-aachen.de>
+#          Alexandre Gramfort <alexandre.gramfort@telecom-paristech.fr>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+from numpy.random import shuffle
+from scipy import stats
+from scipy.signal import hilbert, gaussian, hanning
+
+from ..stats import fdr_correction
+from ..time_frequency import morlet
+from ..time_frequency.tfr import cwt
+from ..preprocessing.peak_finder import peak_finder
+from ..filter import band_pass_filter
+from ..parallel import parallel_func
+from ..utils import logger, check_random_state
+
+
+def _abs_cwt(x, W):
+    """ Aux function """
+    return np.abs(cwt(x, [W], use_fft=False).ravel()).astype(np.float)
+
+
+def make_surrogate_data(data):
+    """ Returns a copy of the shuffled surrogate data.
+
+    Parameters
+    ----------
+    data : ndarray, shape (n_epochs, n_times)
+        Data to be shuffled.
+
+    Returns
+    -------
+    shuffled_data : ndarray, shape (n_epochs, n_times)
+        The shuffled data, with each epoch shuffled across n_times.
+    """
+    shuffled_data = data.copy()
+    map(shuffle, shuffled_data)
+    return shuffled_data
```

Comment:
@dengemann do you mean randomizing the phases and not just the time points? Something like this? 

```

    from sklearn.utils import check_random_state
    data = np.asarray(data)
    data_freq = np.fft.rfft(data, axis=0)
    rng = check_random_state(random_state)
    data_freq = np.abs(data_freq) * np.exp(1j * rng.random_sample(data_freq.shape) * 2 * np.pi)
    return np.fft.irfft(data_freq, data.shape[0], axis=0)
```